### ISSUE TITLE
when autoFixOnSave fails, the file can't be saved

### ISSUE THREAD

**Issue by vhoyer** [neoclide/coc-eslint#144]
I vote the file to be saved even if i have thrown an error, less disturbing to the flow

**Comment by bartels** [comment]
This is a problem for me as well. I have autoFixOnSave enabled since I usually do want to use it. But some projects do not have an eslint configuration because they don't use eslint at all. In these cases not only do I get a lound error message but the file cannot be saved.

I understand that this feature does require an eslint configuration to work, since that's what eslint itself requires. But it should probably not prevent writing files altogether

```
Error detected while processing BufWritePre Autocommands for "*"..function coc#rpc#request[4]..<SNR>45_request[34]..BufWritePre Autocommands for "*"..function coc#rpc#request[4]..<SNR>45_request:
line   31:
E605: Exception not caught: Error on request: Vim(return):Error invoking 'doAutocmd' on channel 3 (coc):^@Request textDocument/codeAction failed with message: No ESLint configuration found in /home/mrmelve/.dotfiles/vim/.vim.
```
Maybe there is a way to catch and show the warning message but still allow the autocmd to finish successfully?

**Comment by ic-768** [comment]
Same issue

**Comment by ensia96** [comment]
To avoid this error, I think coc-eslint needs default eslint config that can be used when there is no config files in project.

I changed my formatter to [prettier](https://github.com/neoclide/coc-prettier) to temporarily fix this issue for my own.

I added _ to my old config property `"eslint.autoFixOnSave"` and add `"coc.preferences.formatOnSaveFileTypes": ["*"]`

```json
{
  "_eslint.autoFixOnSave": true,
  ...
  "coc.preferences.formatOnSaveFileTypes": ["*"],
  ...
}
```
